pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -371,7 +368,7 @@ stdenv.mkDerivation (finalAttrs: {
   # when cross-compiling.
   + ''
     shopt -s extglob
-    patchShebangs tools test src/!(rpm|kernel-install|ukify) src/kernel-install/test-kernel-install.sh
+    patchShebangs tools test src/!(rpm|ukify) src/kernel-install/test-kernel-install.sh
```

Comment:
What was the reason for this change?  It broke cross compilation (because these scripts end up with a build bash shebang), and isn't mentioned in the commit message.

Reply:
My bad for not making this explicit. But for all I know, `kernel-install` is not a script anymore. It was rewritten in C. What am I missing?

Reply:
[All the .install shell scripts.](https://github.com/systemd/systemd/tree/main/src/kernel-install)

Reply:
I am seeing
```
error: build of '/nix/store/llnwcn2bgri0m9cn58h8ym6wj42s1jyv-systemd-minimal-aarch64-unknown-linux-gnu-254.3.drv' on 'ssh://build01-x86-newtype' failed: output '/nix/store/7p5kh7smhwni9pwihs3r4mnzm5sxr022-systemd-minimal-aarch64-unknown-linux-gnu-254.3' is not allowed to refer to the following paths:
         /nix/store/vqvj60h076bhqj6977caz0pfxs6543nb-bash-5.2-p15
```

Reply:
Could it be related to this: https://github.com/NixOS/nixpkgs/pull/252874 ?

Reply:
Applying these patches seems to fix it.

Reply:
So we need to remove the bash path in some out thingie

Reply:
No, this line just needs to be put back as it was before since the .install scripts get installed, as Alyssa mentioned, and need to use the hosr bash, not the build bash

Reply:
I'll open a PR

Reply:
https://github.com/NixOS/nixpkgs/pull/255165

Reply:
Thank you @alyssais for the ping and sorry for the random drop

Reply:
(and thank you @nikstur for fast turnaround in an ICE)